ogr/ogrsf_frmts/miramon/ogrmiramonlayer.cpp

```diff
@@ -0,0 +1,2613 @@
+        /* ---------------------------------------------------------------- */
+        const char *pszVersion = CSLFetchNameValue(papszOpenOptions, "Version");
+        int nMMVersion;
+
+        if (pszVersion)
+        {
+            if (EQUAL(pszVersion, "V1.1"))
+                nMMVersion = MM_32BITS_VERSION;
+            else if (EQUAL(pszVersion, "V2.0") ||
+                     EQUAL(pszVersion, "last_version"))
+                nMMVersion = MM_64BITS_VERSION;
+            else
+                nMMVersion = MM_32BITS_VERSION;  // Default
+        }
+        else
+            nMMVersion = MM_32BITS_VERSION;  // Default
+
+        /* ---------------------------------------------------------------- */
+        /*      Establish the charset of the .dbf files                     */
+        /* ---------------------------------------------------------------- */
+        const char *pszdbfEncoding =
+            CSLFetchNameValue(papszOpenOptions, "DBFEncoding");
+        char nMMRecode;
+
+        if (pszdbfEncoding)
+        {
+            if (EQUAL(pszdbfEncoding, "UTF8"))
+                nMMRecode = MM_RECODE_UTF8;
+            else  //if (EQUAL(pszdbfEncoding, "ANSI"))
+                nMMRecode = MM_RECODE_ANSI;
+        }
+        else
+            nMMRecode = MM_RECODE_ANSI;  // Default
+
+        /* ----------------------------------------------------------------- */
+        /*      Establish the nMemoryRatio to use                            */
+        /* ----------------------------------------------------------------- */
+        const char *pszMemoryRatio =
+            CSLFetchNameValue(papszOpenOptions, "CreationMemoryRatio");
+
+        if (pszMemoryRatio)
+            nMMMemoryRatio = atof(pszMemoryRatio);
+        else
+            nMMMemoryRatio = 1;  // Default
+
+        /* ----------------------------------------------------------------- */
+        /*   Establish the descriptors language when                         */
+        /*   creating .rel files                                             */
+        /* ----------------------------------------------------------------- */
+        const char *pszLanguage =
+            CSLFetchNameValue(papszOpenOptions, "CreationLanguage");
+        char nMMLanguage;
+
+        if (pszLanguage)
+        {
+            if (EQUAL(pszLanguage, "CAT"))
+                nMMLanguage = MM_CAT_LANGUAGE;
+            else if (EQUAL(pszLanguage, "SPA"))
+                nMMLanguage = MM_SPA_LANGUAGE;
+            else
+                nMMLanguage = MM_ENG_LANGUAGE;
+        }
+        else
+            nMMLanguage = MM_DEF_LANGUAGE;  // Default
+
+        /* ---------------------------------------------------------------- */
+        /*      Preparing to write the layer                                */
+        /* ---------------------------------------------------------------- */
+        if (!STARTS_WITH(pszFilename, "/vsistdout"))
```

Comment:
Just drop that /vsistdout related test, and de-indent the below block

Reply:
Done